File: BasicVrPlayback/src/main/java/com/bitmovin/player/samples/vr/basic/MainActivity.java
package com.bitmovin.player.samples.vr.basic;

import android.os.Bundle;
import android.view.Window;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.core.view.WindowInsetsControllerCompat;

import com.bitmovin.player.PlayerView;
import com.bitmovin.player.api.Player;
import com.bitmovin.player.api.source.SourceConfig;
import com.bitmovin.player.api.source.SourceType;
import com.bitmovin.player.api.vr.VrConfig;
import com.bitmovin.player.api.vr.VrContentType;

public class MainActivity extends AppCompatActivity {
    private PlayerView bitmovinPlayerView;
    private Player player;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        EdgeToEdge.enable(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.root), (view, insets) -> {
            Insets bars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            view.setPadding(bars.left, bars.top, bars.right, bars.bottom);
            return WindowInsetsCompat.CONSUMED;
        });

        Window window = getWindow();
        WindowInsetsControllerCompat insetsController = WindowCompat.getInsetsController(window,
                window.getDecorView());
        insetsController.setAppearanceLightStatusBars(true);
        insetsController.setAppearanceLightNavigationBars(true);

        bitmovinPlayerView = findViewById(R.id.bitmovinPlayerView);
        player = bitmovinPlayerView.getPlayer();

        // Enabling the gyroscopic controlling for the 360° video
        player.getVr().setGyroscopeEnabled(true);

        initializePlayer();
    }

    @Override
    protected void onStart() {
        bitmovinPlayerView.onStart();
        super.onStart();
    }

    @Override
    protected void onResume() {
        super.onResume();
        bitmovinPlayerView.onResume();
    }

    @Override
    protected void onPause() {
        bitmovinPlayerView.onPause();
        super.onPause();
    }

    @Override
    protected void onStop() {
        bitmovinPlayerView.onStop();
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        bitmovinPlayerView.onDestroy();
        super.onDestroy();
    }

    protected void initializePlayer() {
        // Create a new SourceItem
        SourceConfig vrSourceItem = new SourceConfig("https://cdn.bitmovin.com/content/assets/playhouse-vr/mpds/105560.mpd", SourceType.Dash);

        // Get the current VRConfiguration of the SourceItem
        VrConfig vrConfiguration = vrSourceItem.getVrConfig();
        // Set the VrContentType on the VRConfiguration
        vrConfiguration.setVrContentType(VrContentType.Single);
        // Set the start position to 180 degrees
        vrConfiguration.setStartPosition(180);

        // load source using the created source item
        player.load(vrSourceItem);
    }
}

File: BasicVrPlaybackKotlin/src/main/java/com/bitmovin/player/samples/vr/basic/MainActivity.kt
package com.bitmovin.player.samples.vr.basic

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.updatePadding
import com.bitmovin.player.api.Player
import com.bitmovin.player.api.source.SourceConfig
import com.bitmovin.player.api.source.SourceType
import com.bitmovin.player.api.vr.VrConfig
import com.bitmovin.player.api.vr.VrContentType
import com.bitmovin.player.samples.vr.basic.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var player: Player
    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(binding.root) { view, windowInsets ->
            val insets = windowInsets.getInsets(WindowInsetsCompat.Type.systemBars())
            view.updatePadding(insets.left, insets.top, insets.right, insets.bottom)
            WindowInsetsCompat.CONSUMED
        }
        WindowCompat.getInsetsController(window, window.decorView).apply {
            isAppearanceLightStatusBars = true
            isAppearanceLightNavigationBars = true
        }

        player = binding.playerView.player!!

        // Enabling the gyroscopic controlling for the 360° video
        player.vr.isGyroscopeEnabled = true
        initializePlayer()
    }

    override fun onStart() {
        super.onStart()
        binding.playerView.onStart()
    }

    override fun onResume() {
        super.onResume()
        binding.playerView.onResume()
    }

    override fun onPause() {
        binding.playerView.onPause()
        super.onPause()
    }

    override fun onStop() {
        binding.playerView.onStop()
        super.onStop()
    }

    override fun onDestroy() {
        binding.playerView.onDestroy()
        super.onDestroy()
    }

    private fun initializePlayer() {
        // Create a new SourceItem
        val vrSourceItem = SourceConfig(
                url = "https://cdn.bitmovin.com/content/assets/playhouse-vr/mpds/105560.mpd",
                type = SourceType.Dash,
                vrConfig = VrConfig(
                        vrContentType = VrContentType.Single,
                        startPosition = 180.0
                )
        )

        // load source using the created source item
        player.load(vrSourceItem)
    }
}
